torch/_dynamo/variables/builder.py

```diff
@@ -935,29 +933,30 @@ def _clone_input(value):
 @dataclasses.dataclass
 class TrackedFake:
     fake: Union[FakeTensor, SymInt]
-    sname: str
+    source: Source
 
 
 def wrap_to_fake_tensor_and_record(
-    e, tx, ignore_subclass=False, *, sname: str, static_shapes=False, is_tensor: bool
+    e, tx, ignore_subclass=False, *, source: Optional[Source], is_tensor: bool
 ):
     if type(e) in (torch.Tensor, torch.nn.Parameter) or (
         ignore_subclass and isinstance(e, torch.Tensor)
     ):
-        static_shapes = static_shapes or config.dynamic_shapes is False
-        if type(e) is torch.nn.Parameter:
-            # Always static for params
-            static_shapes = True
+        static_shapes = (
+            source is None
+            or type(e) is torch.nn.Parameter
```

Comment:
Why is and not isinstance here?

Reply:
If it is a Parameter subclass then we need to preserve the particular subclass, but now we have no guarantees that the subclass isn't doing funny business on operations. With Parameter we know what the semantics are. (Also, this is a preexisting condition.)